packages/common/base/src/utils/apiKey/validateAPIKeyPrefix.test.ts

```diff
@@ -0,0 +1,115 @@
+import { describe, expect, test } from "vitest";
+
+import { validateAPIKeyPrefix } from "./validateAPIKeyPrefix";
+
+describe("validateAPIKeyPrefix", () => {
+    test("Should disallow API keys with old format", () => {
+        const prefixes = ["sk_live", "sk_test"];
+
+        for (const prefix of prefixes) {
+            const result = validateAPIKeyPrefix(prefix);
+            if (result.isValid) {
+                throw new Error("Expected API key to be invalid");
+            }
+
+            expect(result.isValid).toBe(false);
+            expect(result.message).toBe(
+                "Old API key format detected. Please create a new API key from the Crossmint console."
+            );
+        }
+    });
+
+    test("Should disallow when usage prefix is invalid", () => {
+        const result = validateAPIKeyPrefix("bk_");
+        if (result.isValid) {
+            throw new Error("Expected API key to be invalid");
+        }
+
+        expect(result.isValid).toBe(false);
+        expect(result.message).toBe("Malformed API key. Must start with 'ck' or 'sk'.");
+    });
+
+    test("Should disallow when API key does not have expected usage origin", () => {
+        const keys = ["ck_development", "sk_development"];
+
+        for (const key of keys) {
+            const usageOrigin = key.startsWith("ck") ? "client" : "server";
+            const expectedUsageOrigin = usageOrigin === "client" ? "server" : "client";
+
+            const result = validateAPIKeyPrefix(key, { usageOrigin: expectedUsageOrigin });
+            if (result.isValid) {
+                throw new Error("Expected API key to be invalid");
+            }
```

Comment:
this shouldn't be needed–`expect(result.isValid).toBe(false);` will take care of this

Reply:
its mostly a ts thing, jest doesnt know result.message exists if u dont do the if
![image](https://github.com/Crossmint/crossmint-sdk/assets/93682696/304f19d0-a1d6-43e8-83bc-a9cd43a28f27)